orttraining/orttraining/python/training/ortmodule/_graph_execution_manager.py

```diff
@@ -321,6 +321,14 @@ def _get_exported_model(self, input_schema: _ModelInputOutputSchemaType, *inputs
             # FlattenedModule needs _InputInfo to expand user input from *args to *args + **kwargs
             self._flattened_module._input_info = self._input_info
 
+            # Leverage cached model if available
+            cache_dir = os.getenv("ORT_CACHE_DIR", default=None)
+            if cache_dir and os.path.exists(cache_dir):
+                filename = os.path.join(cache_dir, "model.onnx")
```

Comment:
Shall we save the model as `f"{hash(self._flattened_model)}.onnx"`?

Reply:
`hash(self._flattened_model)` does not come out to a unique value so my current implementation doesn't work with this naming convention

Reply:
Do you think we could instead do a hash of the model architecture?

like `hash(str(self._flattened_model))`?

Reply:
This is what `str(self.flattened_module)` returns. Is this the correct thing to hash?

`_FlattenedModule(  (_original_module): Net(    (fc): Linear(in_features=10, out_features=1, bias=True)  ))`